### PR TITLE
Set races and urm columns directly.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This PR changes the `User` model so that the `races` and `urm` columns are the source of truth rather than the `races` serialized attribute. Unfortunately, this requires disabling the `sanitize_and_set_race_data` method, as (a) `self.races` is overloaded and (b) `before_save` hooks cannot save the object. As noted in the TODO's, cleaning this up will also follow.

After this PR reaches all environments, I'll follow with a script to remove instances of the `races` serialized attribute, followed by removing the keyword from the list of serialized attributes.

Part of this was erroneously removed in #15465, see the [discussion](https://github.com/code-dot-org/code-dot-org/pull/15465#discussion_r119726195) there.

LOCAL TESTING:
- [x] Confirm that race choices are persisted correctly.
- [x] Confirm that nonsense race choices are sanitized correctly.
- [x] Confirm that closed_dialog and opt_out choices are persisted correctly.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Let me know if you would rather me dismantle the `before_save` hook as part of this PR or in another follow-up PR.

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -568,6 +566,23 @@ def username_required?
     provider == User::PROVIDER_MANUAL
   end
 
+  def update_without_password(params, *options)
+    if params[:races]
+      update_columns(races: params[:races].join(','))
+      if params[:races].include? 'closed_dialog'
+        update_columns(races: 'closed_dialog')
+      end
+      if params[:races].length > 5
```
Comment (ewjordan) [inline dashboard/app/models/user.rb:575]: Presumably the nonsense-filter can or will be removed from where it was previously?

Comment (ashercodeorg) [reply dashboard/app/models/user.rb:575]: This PR is decoupling the serialized attribute from the new columns, so as to allow removal of the former. As such, I'm currently keeping the sanitization in both places (`update_without_password` does the columns, `sanitize_and_set_race_data` does the serialized attribute). As part of deprecating the serialized attribute, the latter will be removed.

**Comment by ashercodeorg** [comment]
It has. After the PR got merged that established writing to both places, I ran the script to backfill old data.

**Comment by ashercodeorg** [comment]
Confirming that, I'm seeing 2037 instances on production that remain to be backfill. Let me rerun the script, I'd guess that those 2K users filled out the race information while the script was running.

**Comment by ashercodeorg** [comment]
Actually, I'm blaming the 2K users on missing `with_deleted` in the backfill script. Fixed on GitHub in #15772. Fixing production data now.

**Pull request merged**